Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -82,44 +116,65 @@ final class SchemesGenerator: SchemesGenerating {
                               testAction: generatedTestAction,
                               launchAction: generatedLaunchAction,
                               profileAction: generatedProfileAction,
-                              analyzeAction: schemeAnalyzeAction(for: project),
+                              analyzeAction: generatedAnalyzeAction,
                               archiveAction: generatedArchiveAction)
+                              
         try scheme.write(path: schemePath.path, override: true)
     }
-
-    /// Returns the build action for the project scheme.
+    
+    /// Generates the scheme build action.
     ///
     /// - Parameters:
-    ///   - project: Project manifest.
-    ///   - generatedProject: Generated Xcode project.
-    ///   - graph: Dependencies graph.
+    ///   - scheme: Scheme manifest.
+    ///   - graph: Tuist graph.
+    ///   - rootPath: Path to the project or workspace.
+    ///   - generatedProjects: Project paths mapped to generated projects.
     /// - Returns: Scheme build action.
-    func projectBuildAction(project: Project,
-                            generatedProject: GeneratedProject,
-                            graph: Graphing) -> XCScheme.BuildAction {
-        let targets = project.sortedTargetsForProjectScheme(graph: graph)
-        let entries: [XCScheme.BuildAction.Entry] = targets.map { (target) -> XCScheme.BuildAction.Entry in
+    func schemeBuildAction(scheme: Scheme,
+                           graph: Graphing,
+                           rootPath: AbsolutePath,
+                           generatedProjects: [AbsolutePath: GeneratedProject]) throws -> XCScheme.BuildAction? {
+        guard let buildAction = scheme.buildAction else { return nil }
 
-            let pbxTarget = generatedProject.targets[target.name]!
-            let buildableReference = targetBuildableReference(target: target,
+        let buildFor: [XCScheme.BuildAction.Entry.BuildFor] = [
+            .analyzing, .archiving, .profiling, .running, .testing,
+        ]
+
+        var entries: [XCScheme.BuildAction.Entry] = []
+        var preActions: [XCScheme.ExecutionAction] = []
+        var postActions: [XCScheme.ExecutionAction] = []
+
+        try buildAction.targets.forEach { buildActionTarget in
+            let pathToProject = buildActionTarget.projectPath
+            guard let target = try graph.target(path: pathToProject,
+                                                name: buildActionTarget.name) else { return }
+            guard let generatedProject = generatedProjects[pathToProject] else { return }
+            guard let pbxTarget = generatedProject.targets[buildActionTarget.name] else { return }
+            let xcodeProjectPath = buildActionTarget.projectPath.appending(component: generatedProject.name)
```

Comment:
minor: we interchangeably use `pathToProject` (`= buildActionTarget.projectPath`) and `buildActionTarget.projectPath` worth being consistent. 

(heh might have been me who introduced this inconsistency! 😅)